### PR TITLE
Add versioning

### PR DESCRIPTION
**Pull request by NicFragale**
Add the option to output a version

### CODE REVIEW

**Comment by NicFragale** [comment]
Will have to add the following for it to be meaningful.

[[tag]] = current version tag for building

```
go build -ldflags "-X main.version=[[tag]] -X main.commit=$(git rev-parse HEAD) -X main.date=$(date -u +%Y-%m-%dT%H:%M:%SZ)"
```

**Comment by NicFragale** [comment]
Addresses https://github.com/openziti-test-kitchen/zssh/issues/43#issue-2523449151

File: zssh/zssh/main.go [hunk 1]
```diff
@@ -33,13 +33,17 @@ import (
...
+	version = "v0.0.0"
+	commit	= "unknown"
+	date	= "unknown"
 )
 
 var rootCmd = &cobra.Command{
 	Use:   fmt.Sprintf("%s %s <remoteUsername>@<targetIdentity>", ExpectedServiceAndExeName, flags.ServiceName),
 	Short: "Z(iti)ssh, Carb-loaded ssh performs faster and stronger than ssh",
 	Long:  "Z(iti)ssh is a version of ssh that utilizes a ziti network to provide a faster and more secure remote connection. A ziti connection must be established before use",
+	Version: fmt.Sprintf("%s (commit: %s, built: %s)", version, commit, date),
```
Comment (dovholuknf) [inline zssh/zssh/main.go]: can you update zscp, too, please? i'm not a fan of all the additional output for the commit hash and date. i'd rather just see the version number. 

Can you make the output look like:
```
1.0.9B (built:2024-09-13T19:26:54Z)
```
or with the hash use the short hash:
```
1.0.9B (built:2024-09-13T19:26:54Z hash:e9df4770)
```

Comment (dovholuknf) [reply zssh/zssh/main.go]: i see you did zscp too - thx

File: zssh/zssh/zssh
Comment (dovholuknf) [inline zssh/zssh/zssh:1]: remove the binary file pls

Comment (NicFragale) [reply zssh/zssh/zssh:1]: Good catch - done!

File: zssh/zssh/main.go (hunk 1 above)
Comment (NicFragale) [reply zssh/zssh/main.go]: image

**Review by dovholuknf — APPROVED** [review]

**Comment by dovholuknf** [comment]
Thanks!

**Pull request merged**